pytheus/metrics.py

```diff
@@ -39,12 +51,24 @@ def __init__(
     def _validate_labels(self, labels: Sequence[str]):
         """
         Validates label names according to the regex.
-        Labels starting with `__` are reserved for internal use.
+        Labels starting with `__` are reserved for internal use by Prometheus.
         """
         for label in labels:
             if label.startswith('__') or label_name_re.fullmatch(label) is None:
                 raise ValueError(f'Invalid label name: {label}')
 
+    def collect(self) -> Sequence[Sample] | Iterator[Sample]:
+        """
+        Collects Samples for the metric.
+        If there are no required labels, will return a sequence with a single Sample.
```

Comment:
I left both for now to make mypy happy, the final one we will decide depends on the implementation of the Registry, so depends on what will be more useful for it. I was opting more for the `Iterator` but it might be better something else.